atsign-foundation/at_rust
# refactor: Full re-write of SDK

**- What I did**
I re-wrote the code from scratch to allow for better extensibility and maintainability (and for fun). There is no additional functionality (yet) from the previous version. The main changes include:
- Breaking the repo down into [workspaces](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html)
- Created traits for crypto and TLS to allow "drop-in" replacements if needed in the future
- Used `RustCrypto` crates for crypto because it is written in pure Rust so should be compatible with more hardware
- Used `rustls` crate for TLS which should have broader compatibility that the previous crate that was being used
- More tests

**- How to verify it**
- Check through the code
- Run the tests (info about how to do this in the `README.md`)
- Run the examples (info about how to do this in the `README.md`) and check the data is read and written correctly and can be accessed by other SDKs

**- Description for the changelog**
Full re-write of SDK
